docs/pages/api.rst

```diff
@@ -1355,42 +1418,36 @@ Signature: ``Collection::partition(callable ...$callbacks);``
 permutate
 ~~~~~~~~~
 
-Find all the permutations of a collection.
+Find all the `permutations <https://en.wikipedia.org/wiki/Permutation>`_ of a collection.
 
 Interface: `Permutateable`_
 
 Signature: ``Collection::permutate(int $size, $value);``
 
 .. code-block:: php
 
-    $collection = Collection::fromIterable(['hello', 'how', 'are', 'you'])
-        ->permutate();
+    $collection = Collection::fromIterable(['a', 'b'])
+        ->permutate(); // [['a', 'b'], ['b', 'a']]
 
 pluck
 ~~~~~
 
 Retrieves all of the values of a collection for a given key.
+Nested values can be retrieved using "dot notation" and the wildcard character ``*``.
 
 Interface: `Pluckable`_
 
 Signature: ``Collection::pluck($pluck, $default = null);``
 
-.. code-block:: php
-
-    $fibonacci = static function ($a = 0, $b = 1): array {
-        return [$b, $a + $b];
-    };
-
-    $collection = Collection::unfold($fibonacci)
-        ->limit(10)
-        ->pluck(0);
+.. literalinclude:: code/operations/pluck.php
+  :language: php
 
 prepend
 ~~~~~~~
 
 Push an item onto the beginning of the collection.
 
-.. warning:: If prepended values overwrite existing values, you might find that this operation doesn't work correctly
+.. warning:: If prepended values overwrite existing values or keys, you might find that this operation doesn't work correctly
              when the collection is converted into an array.
              It's always better to never convert the collection to an array and use it in a loop.
              However, if for some reason, you absolutely need to convert it into an array, then use the
```

Comment:
I realised there are multiple similar warnings like this. I think it would be useful to have an initial section describing how to work with lazyness and with this library to avoid unexpected behaviour, loss of data etc. 

Perhaps in a subsequent PR I will put something together. That section needs to come before the API methods because the principles apply to pretty much every method.

Reply:
Yes, good idea. I already wrote about it here: https://loophp-collection.readthedocs.io/en/latest/pages/usage.html#manipulate-keys-and-values

It could be a starting point?

Reply:
Ah nice, I didn't see that section because I didn't go into all the Advanced usages. Can definitely be a starting point 👍